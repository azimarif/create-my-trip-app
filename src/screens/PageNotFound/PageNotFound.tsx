import { Button, Container, Paper, Typography } from '@mui/material';
import { grey } from '@mui/material/colors';
import { useNavigate } from 'react-router-dom';
import { Strings } from '../../locales/English';
import logo from '../../assets/compass.svg';
import backgroundImage from '../../assets/background.png';
import Body from '../../layout/Body';

function PageNotFound() {
  const navigate = useNavigate();

  return (
    <Body>
      <Paper
        elevation={3}
        sx={{ m: 'auto', backgroundImage: `url(${backgroundImage})` }}
      >
        <img src={logo} alt="404" />
        <Container sx={{ display: 'flex', alignItems: 'center' }}>
          <Container sx={{ py: 2 }}>
            <Typography variant="h3" mb={2} sx={{ textAlign: 'center' }}>
              {Strings.PageNotFoundPage.Dialog.Title}
            </Typography>
            <Typography variant="h6" mb={2} color={grey['700']}>
              {Strings.PageNotFoundPage.Dialog.Description1}
            </Typography>
            <Typography color={grey['700']}>
              {Strings.PageNotFoundPage.Dialog.Description2}
            </Typography>
            <Typography mb={2} color={grey['700']}>
              {Strings.PageNotFoundPage.Dialog.Description3}
            </Typography>
            <Button variant="contained" onClick={() => navigate('/')}>
              {Strings.PageNotFoundPage.Dialog.ActionName}
            </Button>
          </Container>
        </Container>
      </Paper>
    </Body>
  );
}

export default PageNotFound;
