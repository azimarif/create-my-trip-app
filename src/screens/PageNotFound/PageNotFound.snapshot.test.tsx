import { render } from '@testing-library/react';
import PageNotFound from './PageNotFound';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
}));

test('PageNotFound component snapshot', () => {
  const { container } = render(<PageNotFound />);
  expect(container).toMatchSnapshot();
});
