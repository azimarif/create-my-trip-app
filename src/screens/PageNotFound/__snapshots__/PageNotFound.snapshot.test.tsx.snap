// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound component snapshot 1`] = `
<div>
  <div
    class="css-11wj5jt"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1vwoz01-MuiPaper-root"
    >
      <img
        alt="404"
      />
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-wbzd9a-MuiContainer-root"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-18p1znf-MuiContainer-root"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 css-15380h8-MuiTypography-root"
          >
            Looks like we’re lost.
          </h3>
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1wetlb4-MuiTypography-root"
          >
            Sorry, the page you are looking for cannot be found or may have been moved. You can go back to the homepage or use our search.
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ps0fg4-MuiTypography-root"
          >
            Our main destinations:
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-y0t9kg-MuiTypography-root"
          >
            Africa•Asia•Europe•India•Central America•Middle East•North America•Oceania•Polar Regions•South America
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Back to Home
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
